tests/unit_tests/client/address_book/storage.cc

```diff
@@ -0,0 +1,192 @@
+ * 1. Redistributions of source code must retain the above copyright notice, this list of     //
+ *    conditions and the following disclaimer.                                                //
+ *                                                                                            //
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list     //
+ *    of conditions and the following disclaimer in the documentation and/or other            //
+ *    materials provided with the distribution.                                               //
+ *                                                                                            //
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be       //
+ *    used to endorse or promote products derived from this software without specific         //
+ *    prior written permission.                                                               //
+ *                                                                                            //
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY        //
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF    //
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL     //
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,       //
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,               //
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS    //
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,          //
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF    //
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.               //
+ */
+
+#define BOOST_TEST_DYN_LINK
+
+#include <boost/test/unit_test.hpp>
+
+#include <iostream>
+
+#include "client/address_book/impl.h"
+#include "client/address_book/storage.h"
+
+#include "core/router/context.h"
+
+namespace client = kovri::client;
+namespace core = kovri::core;
+
+struct AddressBookStorageFixture
+{
+  /// @alias Subscription
+  /// @brief Subscription alias for readability and convenience
+  /// @details Intended for user-convenience and readability
+  using SubscriptionType = client::AddressBook::SubscriptionType;
+
+  AddressBookStorageFixture() : m_Storage(nullptr)
+  {
+    // Setup temporary data directory
+    core::context.SetCustomDataDir(temp_path.string());
+
+    // Ensure Client and AddressBook paths exist
+    core::EnsurePath(core::GetPath(core::Path::Client));
+
+    core::EnsurePath(core::GetPath(core::Path::AddressBook));
+
+    // Setup storage
+    m_Storage = GetStorageInstance();
+  }
+
+  ~AddressBookStorageFixture()
+  {
+    RemoveFiles();
```

Comment:
```diff
diff --git a/tests/unit_tests/client/address_book/storage.cc b/tests/unit_tests/client/address_book/storage.cc
index b93b8d68..ee27557a 100644
--- a/tests/unit_tests/client/address_book/storage.cc
+++ b/tests/unit_tests/client/address_book/storage.cc
@@ -65,7 +65,7 @@ struct AddressBookStorageFixture
 
   ~AddressBookStorageFixture()
   {
-    RemoveFiles();
+    BOOST_CHECK_NO_THROW(RemoveFiles());
   }
 
   /// @brief Remove files created by AddressBookStorage
@@ -90,6 +90,7 @@ struct AddressBookStorageFixture
       {
         core::Exception ex(__func__);
         ex.Dispatch();
+        throw;
       }
   }
 
```

because we want to know when something goes wrong.

Reply:
Good point, I'll add the exception handling.